client/state/sites/reducer.js

```diff
@@ -214,6 +215,32 @@ export function items( state = {}, action ) {
 
 			return state;
 		}
+
+		case InvitesActionTypes.RECEIVE_INVITE_ACCEPTED_SUCCESS: {
+			const { invite, data } = action;
+
+			if ( ! invite || ! data || ! data.sites ) {
+				return state;
+			}
+
+			if ( invite.role === 'follower' || invite.role === 'viewer' ) {
+				return state;
+			}
+
+			return reduce(
+				data.sites,
+				( memo, site ) => {
+					if ( memo === state ) {
+						memo = { ...state };
+					}
+
+					memo[ site.ID ] = pick( site, VALID_SITE_KEYS );
+
+					return memo;
+				},
+				state
```

Comment:
Also, the initial state for the `reduce` function should be an empty object. We are recreating the sites list from scratch, not just adding info to the existing one.

This code will leave sites that were deleted (i.e., are not a part of the received list) in `state`.

Reply:
Yeah. I was still in the mindset of just adding the one new site, that's why I put `state` as initial value for `reduce`. But there could be a situation in which a site is deleted on backend while you accept invite and that site deletion wouldn't sync to Calypso so alright, I'll update.